sympy/discrete/transforms.py

```diff
@@ -0,0 +1,68 @@
+"""
+Discrete Fourier Transform, Walsh Hadamard Transform,
+Number Theoretic Transform, Zeta Transform, Mobius Transform
+"""
+from __future__ import print_function, division
+
+from sympy.core import S, Symbol, sympify
+from sympy.core.compatibility import as_int, range
+from sympy.core.function import expand, expand_mul
+from sympy.core.numbers import pi, I
+from sympy.functions.elementary.exponential import exp
+from sympy.ntheory import isprime, primitive_root
+from sympy.utilities.iterables import ibin
+
+def _fourier_transform(seq, inverse=False):
+    """
+    Performs the DFT in complex field, by padding zeros as
+    radix 2 FFT, requires the number of sample points to be
+    a power of 2.
+    """
+
+    if not hasattr(seq, '__iter__'):
+        raise TypeError("Expected a sequence of numeric coefficients " +
+                        "for Fourier Transform")
+
+    a = sympify(seq)
+    if any(x.has(Symbol) for x in a):
+        raise ValueError("Expected non-symbolic coefficients")
+
+    n = len(a)
+    if n < 2:
+        return
+
+    while n&(n - 1):
+        n += n&-n
```

Comment:
I find this difficult to follow. I would rather use something like this:

    if n & (n-1):  # not a power of 2
        n = 2**n.bit_length()

Reply:
Actually `n&-n` masks all bits but the LSB of `n`. So, the carry is propogated until there is only 1 bit set. It is `O(k)`  - `k` is the set bits in `n`.
It is an optimized version of the following:
```
while n&(n - 1):
   n += 1 
```

I haven't run tests but I think it will be faster than calculating `n.bit_length()` which will always be linear in the position of MSB. If you want I can benchmark them both.

Reply:
I don't think it would be necessary to optimize this. The time will be nanoseconds anyway.

Reply:
Okay, will make the changes if you think it is more readable.